sight_service/worklist_scheduler_opt.py

```diff
@@ -61,53 +58,52 @@ def launch(self,
   @overrides
   def propose_action(
       self, request: service_pb2.ProposeActionRequest
-  ) -> service_pb2.ProposeActionResponse:
+    ) -> service_pb2.ProposeActionResponse:
     # print('request in propose actions: ', request)
 
     attributes = param_proto_to_dict(request.attributes)
     action_attrs = param_proto_to_dict(request.action_attrs)
 
-    with self.pending_lock.gen_wlock():
-      self.pending_samples[self.unique_id] = [action_attrs, attributes]
+    unique_id = self.queue.push_message(message={
+        "action_atrrs": action_attrs,
+        "attributes": attributes
+    })
 
-    # print('self.pending_samples : ',
-    #       self.pending_samples)
-    # print('self.active_samples : ',
-    #       self.active_samples)
-    # print('self.completed_samples : ',
-    #       self.completed_samples)
-    print('self.unique_id : ', self.unique_id)
+    logging.info("self.queue => %s", self.queue)
 
-    # Create response
-    response = service_pb2.ProposeActionResponse(action_id=self.unique_id)
-    self.unique_id += 1
+    response = service_pb2.ProposeActionResponse(action_id=unique_id)
     return response
 
   @overrides
   def GetOutcome(
       self,
       request: service_pb2.GetOutcomeRequest) -> service_pb2.GetOutcomeResponse:
-    # print('self.pending_samples : ',
-    #       self.pending_samples)
-    # print('self.active_samples : ',
-    #       self.active_samples)
-    # print('self.completed_samples : ',
-    #       self.completed_samples)
-    with self.completed_lock.gen_rlock():
-      completed_samples = self.completed_samples
-    with self.pending_lock.gen_rlock():
-      pending_samples = self.pending_samples
-    with self.active_lock.gen_rlock():
-      active_samples = self.active_samples
+
+    logging.info('self.queue => %s', self.queue)
```

Comment:
can remove comment. i'll be noisy

Reply:
which comment ? , if you are talking about logging the queue , it's for debug purpose

Reply:
its keep it for some time , will remove them after stable release

Reply:
if you are referring to something else , please elaborate , thanks

Reply:
Yes, sorry. I meant the logging statement. The queue will likely get long so printing it every time will produce a huge log and will slow down the server. Please comment it out before submitting so it is not in the deployed server.

Reply:
sure , thanks !!